torchgen/dest/register_dispatch_key.py

```diff
@@ -603,6 +608,8 @@ def gen_class(
                 guard_field = "c10::cuda::OptionalCUDAGuard guard_;"
         elif self.backend_index.dispatch_key == DispatchKey.CompositeExplicitAutograd:
             guard_field = "c10::OptionalDeviceGuard guard_;"
+        elif self.backend_index.dispatch_key == DispatchKey.MPS:
+            guard_field = "c10::OptionalDeviceGuard guard_;"
```

Comment:
if you defined a c10::mps::OptionalMPSGuard, it would be a marginal optimization to use it directly here. (no big deal though)

Reply:
This is done @ezyang